invenio_vocabularies/datastreams/readers.py

```diff
@@ -80,7 +79,12 @@ def _iter(self, fp, *args, **kwargs):
     def read(self, item=None, *args, **kwargs):
         """Opens a tar archive or uses the given file pointer."""
         if item:
-            yield from self._iter(fp=item, *args, **kwargs)
+            if isinstance(item, tarfile.TarFile):
+                yield from self._iter(fp=item, *args, **kwargs)
+            else:
+                # If the item is not already a TarFile (e.g. if it is a BytesIO), try to create a TarFile from the item.
+                with tarfile.open(mode=self._mode, fileobj=item) as archive:
```

Comment:
For `ZipFile`, the constructor could take filenames and `BytesIO` regardless.
Here, if we wanted to also support filenames, we would need to do an extra check and call `tarfile.open` differently (not using the `fileobj` parameter) like so:

```python
elif isinstance(item, os.PathLike):
    # If the item is not already a TarFile (e.g. if it is a PathLike), try to create a TarFile from the item.
    with tarfile.open(item, mode=self._mode) as archive:
        yield from self._iter(fp=archive, *args, **kwargs)
```

However, for now our only use case is getting a `BytesIO`, so I would say that we can keep it like this for now.